### PR TITLE
Convert some tests off of --nominal

### PR DESCRIPTION
**Pull request by tlively**
In preparation to remove the nominal type system, which is nonstandard and not
usable for modules with nontrivial external linkage requirements, port an
initial batch of tests to use the standard isorecursive type system.

The port involves reordering input types to ensure that supertypes precede their
subtypes and inserting rec groups to ensure that structurally identical types
maintain their separate identities.

More tests will be ported in future PRs before the nominal type system is
removed entirely.

### CODE REVIEW

**Comment by tlively** [comment]
Current dependencies on/for this PR:
* main
  * **PR #5660** <a href="https://app.graphite.dev/github/pr/WebAssembly/binaryen/5660" target="_blank">Graphite</a>  👈

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/WebAssembly/binaryen/5660?utm_source=stack-comment).

File: test/lit/parse-nominal-types-extends.wast [hunk 1]
```diff
@@ -1,62 +1,66 @@
...
   ;; CHECK:      (type $super (func))
   (type $super (func))
 
-  ;; CHECK:      (global $g (ref null $super) (ref.null nofunc))
-  (global $g (ref null $super) (ref.null $sub))
+  ;; CHECK:      (type $sub (func_subtype $super))
+  (type $sub (func) (extends $super))
+
+  ;; CHECK:      (global $g (ref null $sub) (ref.null nofunc))
+  (global $g (ref null $sub) (ref.null nofunc))
```
Comment (kripken) [inline test/lit/parse-nominal-types-extends.wast:17]: I understand why the types are reordered, but why is the global's type changed?

Comment (tlively) [reply test/lit/parse-nominal-types-extends.wast:17]: I noticed that `$sub` wasn't appearing in the output, which was probably an unintended consequence of changing `ref.null` to use bottom types. Changing the type of the global ensures both types appear in the output.

**Review by kripken — APPROVED** [review]

**Comment by kripken** [comment]
This PR appears to have introduced a test update problem. When I pull main, build, and update tests, I get a diff, starting with this:
```diff
diff --git a/test/lit/heap-types.wast b/test/lit/heap-types.wast
index 9f0e43602..8a97f1748 100644
--- a/test/lit/heap-types.wast
+++ b/test/lit/heap-types.wast
@@ -105,17 +105,9 @@
 )
 
 (module
-  ;; CHECK:      (type $vector (array (mut f64)))
   ;; NOMNL:      (type $vector (array (mut f64)))
+  ;; CHECK:      (type $vector (array (mut f64)))
   (type $vector (array (mut f64)))
-  ;; CHECK:      (func $test (type $ref|$vector|_i32_f64_i32_=>_none) (param $ref (ref $vector)) (param $index i32) (param $value f64) (param $size i32)
-  ;; CHECK-NEXT:  (array.fill $vector
-  ;; CHECK-NEXT:   (local.get $ref)
-  ;; CHECK-NEXT:   (local.get $index)
-  ;; CHECK-NEXT:   (local.get $value)
-  ;; CHECK-NEXT:   (local.get $size)
-  ;; CHECK-NEXT:  )
-  ;; CHECK-NEXT: )
   ;; NOMNL:      (func $test (type $ref|$vector|_i32_f64_i32_=>_none) (param $ref (ref $vector)) (param $index i32) (param $value f64) (param $size i32)
   ;; NOMNL-NEXT:  (array.fill $vector
   ;; NOMNL-NEXT:   (local.get $ref)
@@ -124,6 +116,14 @@
   ;; NOMNL-NEXT:   (local.get $size)
   ;; NOMNL-NEXT:  )
   ;; NOMNL-NEXT: )
+  ;; CHECK:      (func $test (type $ref|$vector|_i32_f64_i32_=>_none) (param $ref (ref $vector)) (param $index i32) (param $value f64) (param $size i32)
+  ;; CHECK-NEXT:  (array.fill $vector
+  ;; CHECK-NEXT:   (local.get $ref)
+  ;; CHECK-NEXT:   (local.get $index)
+  ;; CHECK-NEXT:   (local.get $value)
+  ;; CHECK-NEXT:   (local.get $size)
+  ;; CHECK-NEXT:  )
+  ;; CHECK-NEXT: )
   (func $test (param $ref (ref $vector))
               (param $index i32)
               (param $value f64)
```
Might be that the vestigial `NOMNL` confuse the update script? Happens in 3 files:
```
diff --git a/test/lit/heap-types.wast b/test/lit/heap-types.wast
diff --git a/test/lit/isorecursive-good.wast b/test/lit/isorecursive-good.wast
diff --git a/test/lit/subtype-chain.wast b/test/lit/subtype-chain.wast
```

**Pull request merged**